python/ray/scripts/scripts.py

```diff
@@ -827,6 +827,13 @@ def stop(force):
             corpus = (proc_cmd
                       if filter_by_cmd else subprocess.list2cmdline(proc_args))
             if keyword in corpus:
+                # This is a way to avoid killing redis server that's not started by Ray.
+                # We are using a simple hacky solution here since
+                # Redis server will anyway removed soon from the ray repository.
+                if (keyword == "redis-server"
+                        and "core/src/ray/thirdparty/redis/src/redis-server"
```

Comment:
this looks a bit fragile too... i.e. it probably won't work with windows, and subject to failure if the path changes.

Reply:
I think we don't need to worry about path changes. Redis will be removed from Ray in a couple months, and this path hasn't changed for a very long time (more than 2 years, this was the same). It is also hard-coded here https://github.com/ray-project/ray/blob/a60251f47a6f675a8ae4fb8ef33534a79f6ac92c/python/setup.py#L139

For Windows, I will add tests and see if it works on Windows